specification/refarch.adoc

```diff
@@ -147,311 +189,315 @@ assigned to the TVM by the VMM.
-want to be aware of the value programmed into stimecmp to guard against 
+The TSM may not need to program stimecmp on its own, though it may verify
+that time is not going back for a TVM. If the TSM needs to start a timer,
+it should context switch the stimecmp CSR and replace it with its timeout
+value if it's later than the timer it wants to start. The TSM may still
+want to be aware of the value programmed into stimecmp to guard against
 step attacks on TVMs.
 
-Any NMIs experienced during TSM/TVM execution are always handled by the 
-TSM-driver and must cause the TEEs to be destroyed (preventing any loss of 
-confidential info via clearing of machine state). The TSM and therefore all 
+Any NMIs experienced during TSM/TVM execution are always handled by the
+TSM-driver and must cause the TEEs to be destroyed (preventing any loss of
+confidential info via clearing of machine state). The TSM and therefore all
 TVMs are prevented from execution after that point.
 
 === TSM and TVM Isolation
 
-TSM (and all TVMs) memory is granted by the host OS/VMM but is isolated 
-(via access-control and/or confidentiality-protection) by the HW and TCB 
-elements. The TSM, TVM and HW isolation methods used must be evident in the 
-attestation evidence provided for the TVM since it identifies the hardware 
+TSM (and all TVMs) memory is granted by the host OS/VMM but is isolated
+(via access-control and/or confidentiality-protection) by the HW and TCB
+elements. The TSM, TVM and HW isolation methods used must be evident in the
+attestation evidence provided for the TVM since it identifies the hardware
 and the TSM-driver.
 
-There are two facets of TVM and TSM memory isolation that are 
+There are two facets of TVM and TSM memory isolation that are
 implementation-specific:
 
-*a)* *Isolation from host software access* -  The CPU may enforce a 
-hardware-based access-control of TSM memory to prevent access from host 
-software (VMM and host OS) V=0, HS-mode untrusted code. TEE and TVM address 
-spaces are identified by an additional (implementation-defined) *Confidential 
-mode qualifier* to maintain the isolation during access and in internal 
-caches, e.g. Hart TLB lookup may be extended with the Confidential mode 
-qualifier. TVM memory isolation must support sparse memory management 
-models and architectural page-sizes of 4KB, 64K, 2MB, 1GB (and optionally 
-512GB). For example, The hardware may provide a memory ownership tracking 
-table where there is an entry per physical page. The memory ownership 
-tracking table may be a radix tree or a flat table. The memory ownership 
-tracking table may allow memory ownership at multiple granularities such as 
-4K, 64K, 2M, 1G, etc. The memory ownership table may be enforced at the 
+*a)* *Isolation from host software access* -  The CPU must enforce
+hardware-based access-control of TSM memory to prevent access from host software
+(VMM and host OS) V=0, HS-mode untrusted code (the hosting supervisor domain).
```

Comment:
```suggestion
hardware-based access-control of TSM memory to prevent access from host supervisor domain components
(VMM and host OS) V=0, HS-mode untrusted code.
```